WcaOnRails/spec/models/team_members_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+describe TeamMember do
+  it "has a valid factory" do
+    expect(FactoryGirl.build(:team_member)).to be_valid
+  end
+
+  it "ensures start date is before end date" do
+    team_member = FactoryGirl.build :team_member, start_date: 1.day.ago, end_date: 2.days.ago
+    expect(team_member).to be_invalid
+    expect(team_member.errors.messages[:start_date]).to eq ["must be earlier than end date"]
+  end
+
+  it "ensures the position is from the committee this membership is associated with" do
+    unrelated_committee_position = FactoryGirl.create :committee_position
+    team_member = FactoryGirl.build :team_member
+    team_member.update_attributes(committee_position_id: unrelated_committee_position.id)
+    expect(team_member).to be_invalid
+    expect(team_member.errors.messages[:committee_position_id]).to eq ["must be a position from the committee this team member is part of"]
+  end
+
+  it "ensures a senior delegate is recognised by the model" do
+    team_member = FactoryGirl.create :team_member, :senior_delegate
+    expect(team_member.senior_delegate?).to eq true
+  end
+
+  it "ensures a user cannot have overlapping memberships to a single team" do
+    team_member = FactoryGirl.create :team_member
+    team_member2 = FactoryGirl.build :team_member, user_id: team_member.user_id, team_id: team_member.team_id, committee_position_id: team_member.committee_position_id
```

Comment:
These two have *the same* dates, so the test doesn't test the actual overlapping, just add dates like from 4 Dec to 20 Dec and 15 Dec to now.